src/user_crate/mod.rs

```diff
@@ -210,12 +211,52 @@ fn parse_source_and_deps(code_and_deps: &str) -> eyre::Result<(syn::Block, toml:
     let user_dependencies: toml::value::Table =
         toml::from_str(&deps_block).map_err(PlRustError::ParsingDependenciesBlock)?;
 
+    if crate::gucs::allow_listed_dependencies_only() {
+        validate_dependences_are_allowed(&user_dependencies);
+    }
+
     let user_code: syn::Block =
         syn::parse_str(&code_block).map_err(PlRustError::ParsingCodeBlock)?;
 
     Ok((user_code, user_dependencies))
 }
 
+#[tracing::instrument(level = "debug", skip_all)]
+fn validate_dependences_are_allowed(user_dependencies: &toml::value::Table) {
+    let allowed_deps = crate::gucs::get_allow_listed_dependencies();
+    let mut unsupported_deps = std::collections::HashMap::<String, toml::value::Value>::new();
+
+    for (dep, version) in user_dependencies {
+        match version {
+            toml::Value::String(ver) => {
+                let req = semver::VersionReq::parse(ver.as_str()).unwrap();
+
+                if allowed_deps.contains_key(dep)
+                    && req.matches(
+                        &semver::Version::parse(&allowed_deps.get(dep).unwrap().as_str().unwrap())
+                            .unwrap(),
+                    )
+                {
+                    // version is supported
+                } else {
+                    unsupported_deps.insert(dep.to_string(), version.clone());
+                }
+            }
+            _ => {
+                pgx::error!("The toml::Value {:?} is not currently supported", version);
+            }
+        }
+    }
+
+    if !unsupported_deps.is_empty() {
+        pgx::error!(
+            "The following dependencies are unsupported {:?}, environment only supports {:?}",
```

Comment:
"Environment" is vague? This can be plainer: The configured PL/Rust permissions don't allow this. That way if someone is the system administrator they know what to change (or not) to fix this.

Reply:
So the reason I wanted to include the existing crates that are supported is that it seems like an easy way to see what's actually there, instead of having to also expose it via documentation/a SQL function

Reply:
Including the contents of the list is fine, I just want the language to be plainer.